src/v/kafka/server/handlers/metadata.cc

```diff
@@ -51,7 +51,6 @@ make_topic_response_from_topic_metadata(model::topic_metadata&& tp_md) {
           p.error_code = error_code::none;
           p.partition_index = p_md.id;
           p.leader_id = p_md.leader_node.value_or(model::node_id(-1));
-          p.leader_epoch = 0;
```

Comment:
Should this be `p.leader_epoch = invalid_leader_epoch`?

Reply:
nice catch. I was assuming invalid leach epoch was defaulted in the code gen, but I don't see it.

Reply:
it is initialized to `-1` in code generated for metadata response.

Reply:
```json
 { "name": "LeaderEpoch", "type": "int32", "versions": "7+", "default": "-1", "ignorable": true,
          "about": "The leader epoch of this partition." },
```